### PR TITLE
Migrate job runs to use run channel

### PR DESCRIPTION
**Pull request by rupurt**
- [x] Make `services.ExecuteRunAtBlock` private
- [x] Remove services.ExecuteRun
- [x] Figure out new way to test error return value from `services.ExecuteRunAtBlock` e.g. `No unfinished tasks to run`
- [x] Start run in channel without block number `services.BeginRun` (Cron, RunAt)
- [x] Start run in channel with block number `services.BeginRunAtBlock` (RunLog, EthLog)

### CODE REVIEW

File: services/job_runner_test.go [hunk 1]
```diff
@@ -219,82 +223,92 @@ func TestJobRunner_ExecuteRun(t *testing.T) {
...
 	}{
-		{models.RunStatusUnstarted, false},
-		{models.RunStatusInProgress, true},
-		{models.RunStatusPendingConfirmations, false},
-		{models.RunStatusPendingSleep, false},
-		{models.RunStatusPendingBridge, false},
-		{models.RunStatusErrored, true},
-		{models.RunStatusCompleted, true},
+		{models.RunStatusUnstarted, models.RunStatusCompleted},
+		{models.RunStatusInProgress, models.RunStatusInProgress},
```
Comment (se3000) [inline services/job_runner_test.go]: Shouldn't `InProgress` eventually become `Completed`?

File: services/job_runner.go [hunk 2]
```diff
@@ -160,24 +160,34 @@ func (rm *jobRunner) workerCount() int {
 	return len(rm.workers)
 }
 
-// BeginRun creates a new run if the job is valid and starts the job.
-func BeginRun(
+// BuildRun checks to ensure the given job has not started or ended before
+// creating a new run for the job.
+func BuildRun(
```
Comment (se3000) [inline services/job_runner.go]: Throwing an idea out there that eventually we move BuildRun*, EnqueueRun*, and a few other methods that don't actually need things out of the `adapters` or `services` package into store, where more packages can use them without getting circular dependencies. Happy to wait until there's a more pressing need for it, but I think it's cool that now all those functions just rely on `models` and `store`. Much cleaner.

File: services/job_runner.go [hunk 3]
```diff
@@ -325,19 +352,19 @@ func startTask(
...
-		return fmt.Errorf("ExecuteRun: Job#%v: %v", run.JobID, err)
+		return fmt.Errorf("executeRunAtBlock: Job#%v: %v", run.JobID, err)
 	}
 	return nil
 }
 
-// JobRunnerError contains the field for the error message.
-type JobRunnerError struct {
+// RecurringScheduleJobError contains the field for the error message.
+type RecurringScheduleJobError struct {
```
Comment (se3000) [inline services/job_runner.go]: 👍

File: services/job_runner.go (hunk 2 above)
Comment (rupurt) [reply services/job_runner.go]: 👍 I'll add it in another PR

**Pull request merged**